plugins/modules/aci_l4l7_policy_based_redirect_destination.py

```diff
@@ -0,0 +1,404 @@
+    policy = module.params.get("policy")
+    ip = module.params.get("ip")
+    additional_ip = module.params.get("additional_ip")
+    mac = module.params.get("mac")
+    logical_device = module.params.get("logical_device")
+    concrete_device = module.params.get("concrete_device")
+    concrete_interface = module.params.get("concrete_interface")
+    destination_name = module.params.get("destination_name")
+    destination_type = module.params.get("destination_type")
+    health_group = module.params.get("health_group")
+    state = module.params.get("state")
+    pod_id = module.params.get("pod_id")
+
+    if not destination_type:
+        destination_type = "l3"
+
+    if destination_type == "l3":
+        aci_class = "vnsRedirectDest"
+        aci_rn = "RedirectDest_ip-[{0}]".format(ip)
+        module_object = ip
+        target_filter = {"ip": ip}
+        child_classes = ["vnsRsRedirectHealthGroup"]
+        redirect_hg_class = "vnsRsRedirectHealthGroup"
+    elif destination_type == "l1/l2":
+        aci_class = "vnsL1L2RedirectDest"
+        aci_rn = "L1L2RedirectDest-[{0}]".format(destination_name)
+        module_object = destination_name
+        target_filter = {"destName": destination_name}
+        child_classes = ["vnsRsL1L2RedirectHealthGroup", "vnsRsToCIf"]
+        redirect_hg_class = "vnsRsL1L2RedirectHealthGroup"
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="vnsSvcRedirectPol",
+            aci_rn="svcCont/svcRedirectPol-{0}".format(policy),
+            module_object=policy,
+            target_filter={"name": policy},
+        ),
+        subclass_2=dict(
+            aci_class=aci_class,
+            aci_rn=aci_rn,
+            module_object=module_object,
+            target_filter=target_filter,
+        ),
+        child_classes=child_classes,
+    )
+    aci.get_existing()
+
+    if state == "present":
+        if destination_type == "l1/l2" and additional_ip is not None:
+            aci.fail_json(msg="You cannot provide an additional_ip when configuring an l1/l2 destination")
+        elif destination_type == "l3" and (logical_device, concrete_device, concrete_interface) != (None, None, None):
+            aci.fail_json(msg="You cannot provide a logical_device, concrete_device or concrete_interface when configuring an l3 destination")
+        elif destination_type == "l1/l2" and (logical_device, concrete_device, concrete_interface) == (None, None, None):
+            aci.fail_json(msg="You must provide a logical_device, concrete_device and concrete_interface when configuring an l1/l2 destination")
+        elif destination_type == "l1/l2" and ip is not None:
+            aci.fail_json(msg="You cannot provide an ip when configuring an l1/l2 destination")
```

Comment:
have you considered creating a dictionary input for l1/l2 and l3, then deducing the type from the dictionary with a mutually exclusive in the argument spec? and adding the dicts to required if for present, that one of the two is required

this would simplify all the error handling and containerise the config on input side

Reply:
We could do this but I didn't want to change the original code too much

Reply:
ok